// ngx-online-status/bundles/ngx-online-status.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common'], factory) :
	(factory((global['ngx-online-status'] = {}),global.ng.core,global.ng.common));
}(this, (function (exports,core,common) { 'use strict';

var OnlineStatusType = {};
OnlineStatusType.OFFLINE = 0;
OnlineStatusType.ONLINE = 1;
OnlineStatusType[OnlineStatusType.OFFLINE] = "OFFLINE";
OnlineStatusType[OnlineStatusType.ONLINE] = "ONLINE";
var OnlineStatusService = /** @class */ (function () {
    function OnlineStatusService() {
        this.status = new core.EventEmitter(true);
        this.bind();
        window.addEventListener(OnlineStatusService.EVENT_TYPE_ONLINE, this.onOnline);
        window.addEventListener(OnlineStatusService.EVENT_TYPE_OFFLINE, this.onOffline);
    }
    /**
     * @return {?}
     */
    OnlineStatusService.prototype.ngOnDestroy = function () {
        window.removeEventListener(OnlineStatusService.EVENT_TYPE_ONLINE, this.onOnline);
        window.removeEventListener(OnlineStatusService.EVENT_TYPE_OFFLINE, this.onOffline);
    };
    /**
     * @return {?}
     */
    OnlineStatusService.prototype.bind = function () {
        this.onOnline = this.onOnline.bind(this);
        this.onOffline = this.onOffline.bind(this);
    };
    /**
     * @return {?}
     */
    OnlineStatusService.prototype.onOnline = function () {
        this.status.emit(OnlineStatusType.ONLINE);
    };
    /**
     * @return {?}
     */
    OnlineStatusService.prototype.onOffline = function () {
        this.status.emit(OnlineStatusType.OFFLINE);
    };
    return OnlineStatusService;
}());
OnlineStatusService.EVENT_TYPE_ONLINE = 'online';
OnlineStatusService.EVENT_TYPE_OFFLINE = 'offline';
OnlineStatusService.decorators = [
    { type: core.Injectable },
];
/**
 * @nocollapse
 */
OnlineStatusService.ctorParameters = function () { return []; };
var OnlineStatusModule = /** @class */ (function () {
    function OnlineStatusModule() {
    }
    return OnlineStatusModule;
}());
OnlineStatusModule.decorators = [
    { type: core.NgModule, args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [],
                providers: [
                    OnlineStatusService
                ]
            },] },
];
/**
 * @nocollapse
 */
OnlineStatusModule.ctorParameters = function () { return []; };

exports.OnlineStatusType = OnlineStatusType;
exports.OnlineStatusService = OnlineStatusService;
exports.OnlineStatusModule = OnlineStatusModule;

Object.defineProperty(exports, '__esModule', { value: true });

})));
