// ngx-online-status/ngx-online-status.es5.js
import { EventEmitter, Injectable, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
var OnlineStatusType = {};
OnlineStatusType.OFFLINE = 0;
OnlineStatusType.ONLINE = 1;
OnlineStatusType[OnlineStatusType.OFFLINE] = "OFFLINE";
OnlineStatusType[OnlineStatusType.ONLINE] = "ONLINE";
var OnlineStatusService = /** @class */ (function () {
    function OnlineStatusService() {
        this.status = new EventEmitter(true);
        this.bind();
        window.addEventListener(OnlineStatusService.EVENT_TYPE_ONLINE, this.onOnline);
        window.addEventListener(OnlineStatusService.EVENT_TYPE_OFFLINE, this.onOffline);
    }
    /**
     * @return {?}
     */
    OnlineStatusService.prototype.ngOnDestroy = function () {
        window.removeEventListener(OnlineStatusService.EVENT_TYPE_ONLINE, this.onOnline);
        window.removeEventListener(OnlineStatusService.EVENT_TYPE_OFFLINE, this.onOffline);
    };
    /**
     * @return {?}
     */
    OnlineStatusService.prototype.bind = function () {
        this.onOnline = this.onOnline.bind(this);
        this.onOffline = this.onOffline.bind(this);
    };
    /**
     * @return {?}
     */
    OnlineStatusService.prototype.onOnline = function () {
        this.status.emit(OnlineStatusType.ONLINE);
    };
    /**
     * @return {?}
     */
    OnlineStatusService.prototype.onOffline = function () {
        this.status.emit(OnlineStatusType.OFFLINE);
    };
    return OnlineStatusService;
}());
OnlineStatusService.EVENT_TYPE_ONLINE = 'online';
OnlineStatusService.EVENT_TYPE_OFFLINE = 'offline';
OnlineStatusService.decorators = [
    { type: Injectable },
];
/**
 * @nocollapse
 */
OnlineStatusService.ctorParameters = function () { return []; };
var OnlineStatusModule = /** @class */ (function () {
    function OnlineStatusModule() {
    }
    return OnlineStatusModule;
}());
OnlineStatusModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [],
                providers: [
                    OnlineStatusService
                ]
            },] },
];
/**
 * @nocollapse
 */
OnlineStatusModule.ctorParameters = function () { return []; };
/**
 * Created by Vadym Yatsyuk on 27.01.18
 */
/**
 * Generated bundle index. Do not edit.
 */
export { OnlineStatusType, OnlineStatusService, OnlineStatusModule };
