// ngx-online-status/ngx-online-status.js
import { EventEmitter, Injectable, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

let OnlineStatusType = {};
OnlineStatusType.OFFLINE = 0;
OnlineStatusType.ONLINE = 1;
OnlineStatusType[OnlineStatusType.OFFLINE] = "OFFLINE";
OnlineStatusType[OnlineStatusType.ONLINE] = "ONLINE";

class OnlineStatusService {
    constructor() {
        this.status = new EventEmitter(true);
        this.bind();
        window.addEventListener(OnlineStatusService.EVENT_TYPE_ONLINE, this.onOnline);
        window.addEventListener(OnlineStatusService.EVENT_TYPE_OFFLINE, this.onOffline);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        window.removeEventListener(OnlineStatusService.EVENT_TYPE_ONLINE, this.onOnline);
        window.removeEventListener(OnlineStatusService.EVENT_TYPE_OFFLINE, this.onOffline);
    }
    /**
     * @return {?}
     */
    bind() {
        this.onOnline = this.onOnline.bind(this);
        this.onOffline = this.onOffline.bind(this);
    }
    /**
     * @return {?}
     */
    onOnline() {
        this.status.emit(OnlineStatusType.ONLINE);
    }
    /**
     * @return {?}
     */
    onOffline() {
        this.status.emit(OnlineStatusType.OFFLINE);
    }
}
OnlineStatusService.EVENT_TYPE_ONLINE = 'online';
OnlineStatusService.EVENT_TYPE_OFFLINE = 'offline';
OnlineStatusService.decorators = [
    { type: Injectable },
];
/**
 * @nocollapse
 */
OnlineStatusService.ctorParameters = () => [];

class OnlineStatusModule {
}
OnlineStatusModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [],
                providers: [
                    OnlineStatusService
                ]
            },] },
];
/**
 * @nocollapse
 */
OnlineStatusModule.ctorParameters = () => [];

/**
 * Created by Vadym Yatsyuk on 27.01.18
 */

/**
 * Generated bundle index. Do not edit.
 */

export { OnlineStatusType, OnlineStatusService, OnlineStatusModule };
